pibooth/pictures/concatenate.py

```diff
@@ -41,170 +100,185 @@ def concatenate_pictures_portrait(pictures, footer_texts, bg_color, text_color,
         matrix.size, (matrix_width, matrix_height)), Image.ANTIALIAS)
+
     final_image = new_image_with_background(final_width, final_height, bg_color)
     final_image.paste(matrix, ((final_width - matrix.size[0]) // 2,
                                (final_height - footer_size - matrix.size[1]) // 2), mask=matrix)
 
     if footer_size:
-        # Text part
-        draw = ImageDraw.Draw(final_image)
-
-        # Footer 1
-        name_font = ImageFont.truetype(footer_fonts[0], int(2 / 3. * footer_size))
-        name_width, name_height = draw.textsize(footer_texts[0], font=name_font)
-        footer_x = (final_width - name_width) // 2
-        footer_y = final_height - footer_size - 100
-        draw.text((footer_x, footer_y), footer_texts[0], text_color, font=name_font)
-
-        # Footer 2
-        date_font = ImageFont.truetype(footer_fonts[-1], int(1 / 3. * footer_size))
-        date_width, date_height = draw.textsize(footer_texts[1], font=date_font)
-        footer_x = (final_width - date_width) // 2
-        footer_y = final_height - footer_size + 300
-        draw.text((footer_x, footer_y), footer_texts[1], text_color, font=date_font)
+        draw_footer_text(final_image, portrait, footer_texts, footer_fonts, footer_size, text_color)
+
+    end = time.time()
+    LOGGER.info("Took {}s to create final image with PIL".format(end - start))
 
     return final_image
 
 
-def concatenate_pictures_landscape(pictures, footer_texts, bg_color, text_color, footer_fonts, inter_width=None):
+def concatenate_pictures_opencv(portrait, pictures, footer_texts, bg_color, text_color, footer_fonts, inter_width=None):
+    """ Merge up to 4 PIL images using opencv to manipulate the images.
     """
-    Merge up to 4 PIL images in landscape orientation.
+    matrix_raw_width, matrix_raw_height, inter_width = get_pics_layout_size(pictures, portrait, inter_width)
+    final_width, final_height, matrix_width, matrix_height, footer_size = get_final_image_dimensions(portrait, footer_texts)
+    offset_generator = get_pics_layout_offset(pictures, portrait, inter_width)
 
-      +-------------+     +-------------+     +-------------+     +---+-+-+-+---+
-      |     +-+     |     |   +-+  +-+  |     | +-+ +-+ +-+ |     |   |1| |2|   |
-      |     |1|     |     |   |1|  |2|  |     | |1| |2| |3| |     |   +-+ +-+   |
-      |     +-+     |     |   +-+  +-+  |     | +-+ +-+ +-+ |     |   +-+ +-+   |
-      |             |     |             |     |             |     |   |3| |4|   |
-      +-------------+     +-------------+     +-------------+     +---+-+-+-+---+
-    """
-    widths, heights = zip(*(i.size for i in pictures))
-
-    # starting here we consider that all the images have the same height and widths
-    if inter_width is None:
-        inter_width = max(heights) // 20
+    start = time.time(); overall_start = start; LOGGER.debug("Creating pictures matrix image with size {}x{}".format(matrix_raw_width, matrix_raw_height))
```

Comment:
use pibooth.utils.timeit() function (several can be nested)

Reply:
Done